tests/suites/test_suite_ssl.function

```diff
@@ -4207,10 +4215,24 @@ void tls13_srv_max_early_data_size(int scenario, int max_early_data_size_arg)
         uint32_t remaining = max_early_data_size -
                              server_ep.ssl.early_data_count;
 
-        /* In case of accepted early data, reach max_early_data_size exactly. */
+        early_data_len = sizeof(buf_write);
+        /*
+         * Adjust the length of next data to write depending on the remaining
+         * number of early data bytes we are allowed to write and if we want
+         * to reach the maximum exactly or not.
+         */
         if (early_data_len >= remaining) {
-            early_data_len = remaining;
-            write_early_data_flag = 0;
+            if (reach_max) {
+                early_data_len = remaining;
+                write_early_data_flag = 0;
+            } else {
+                if (early_data_len == remaining) {
+                    early_data_len /= 2;
+                } else {
+                    early_data_len = remaining + 1;
```

Comment:
This could possibly make `early_data_len > sizeof(buf_write)` depending on the initial values.

Reply:
This has been removed.